src/securemul.rs

```diff
@@ -210,23 +210,26 @@ mod tests {
         ];
 
         // create 3 tasks (1 per helper) that will execute secure multiplication
-        let handles = input.into_iter().zip(participants).zip(ring).map(
-            |((input, participant), helper_ring)| {
-                tokio::spawn(async move {
-                    let ctx = ProtocolContext {
-                        participant: &participant,
-                        helper_ring: &helper_ring,
-                    };
-                    let mut stream = secure_multiply(input, &ctx, start_index);
-
-                    // compute a*b
-                    let _ = stream.next().await.expect("Failed to compute a*b");
-
-                    // compute (a*b)*c and return it
-                    stream.next().await.expect("Failed to compute a*b*c")
-                })
-            },
-        );
+        let handles =
+            input
+                .into_iter()
+                .zip(participants)
+                .zip(ring)
+                .map(|((input, prss), helper_ring)| {
+                    tokio::spawn(async move {
```

Comment:
just so i understand, what's the reason for using `tokio::spawn` here vs just directly using `async move`?

Reply:
There is no technical need to spawn a task, I just wanted to model helpers as "root" tasks spawning futures and also wanted to make sure "spawn" constraints on references (`static + Send + Sync ) do not make it impossible to send contexts across spawn boundaries